api/src/neo4j/queries/search.js

```diff
@@ -351,13 +352,33 @@ LIMIT ${limit}
     fetchCompartments({ ids: ids["Compartment"], model, version: v, includeCounts: true }),
   ]);
 
-  // formatting for simple (gem browser) search
+  const resObj = {compartmentalizedMetabolites,
+    metabolites,
+    genes,
+    reactions,
+    subsystems,
+    compartments}
+    const resWithScore = {};
+    for (const [component, result] of Object.entries(resObj)) {
+        if (result) {
+            resWithScore[component] = result.map(obj => {
+                    const objWithScore = {
+                        ...obj,
+                        score: idsToScore[obj.id]
+                    }
+                    return objWithScore;
+                })
+        } else {
+            resWithScore[component] = []
+        }
```

Comment:
Is it really possible have no scores?

Reply:
If there are no result returned from fetch[Component] query (line 347-354), then an empty array should be returned as the search result for that component. So this replaces the previous return statement that returned empty array if no result

```
return {
    metabolite: [...compartmentalizedMetabolites || [], ...metabolites || []],
    gene: genes || [],
    reaction: reactions || [],
    subsystem: subsystems || [],
    compartment: compartments || [],
    }
```

Valid or is it too "off context" to have it within the resWithScore creation?

Reply:
I can't think of any way to refactor this right now 😊